### PR TITLE
webgpu: tidy build script

### PR DESCRIPTION
**Pull request by edevil**
Add compile time flags, include only the needed tint header.

### CODE REVIEW

File: build/BUILD.dawn [hunk 1]
```diff
@@ -1020,6 +1028,8 @@ cc_library(
         "src/dawn/wire/client/ClientInlineMemoryTransferService.cpp",
         "src/dawn/wire/client/Device.cpp",
         "src/dawn/wire/client/Device.h",
+        "src/dawn/wire/client/EventManager.cpp",
```
Comment (fhanau) [inline build/BUILD.dawn:1031]: Looked into the build file a bit more – `dawn_wire` (and by extension `generate_wire`) is currently unused right? Feel free to keep it if we plan to use it in the future.

Comment (edevil) [reply build/BUILD.dawn:1031]: Yes, they are unused at the moment. However, the plan is to have an option of using an external process to access the GPU, and when that happens this wire protocol is what will be used to communicate between these two processes.

I have began work on this but the various CODE * initiatives have taken priority...

File: build/BUILD.dawn [hunk 2]
```diff
@@ -920,7 +926,7 @@ cc_library(
         # Skia has its own vulkan headers and we do not want these to
         # interfere/override those.
         "-Iexternal/vulkan_headers/include",
-    ],
+    ] + COPTS,
     defines = [
```
Comment (fhanau) [inline build/BUILD.dawn:930]: Use local_defines here (regular defines is still needed for dawn_native_macos since objc_library doesn't have local_defines)

**Comment by fhanau** [comment]
@jasnell Since you have lots of JSG experience – defining `-fno-rtti -fno-exceptions` for dawn should be safe as long as dawn types are not exposed to JSG (or other interfaces using RTTI) directly, right?

**Comment by jasnell** [comment]
> defining -fno-rtti -fno-exceptions for dawn should be safe as long as dawn types are not exposed to JSG (or other interfaces using RTTI) directly, right?

Correct.

**Comment by fhanau** [comment]
> > defining -fno-rtti -fno-exceptions for dawn should be safe as long as dawn types are not exposed to JSG (or other interfaces using RTTI) directly, right?
> 
> Correct.

Thanks for confirming! I was looking at enabling this in more places too, got to like 8MB of savings on the workerd debug binary. Verifying that it's safe is not always trivial so that's mostly experimental for now.

**Review by fhanau — APPROVED** [review]

**Comment by fhanau** [comment]
> > safe as long as dawn types are not exposed to JSG (or other interfaces using RTTI) directly, right?
> 
> I do pass around some `jsg:Ref<DAWN TYPE>`s. What does "exposed to JSG" mean in this context? :)

I would assume that's fine as long as those types are not exposed to the JS API (i.e. via `EW_WEBGPU_ISOLATE_TYPES`). If they are wrapped in a struct where RTTI is defined (e.g. GPUComputePassEncoder containing wgpu::ComputePassEncoder) that's also fine (?) If we're unsure, maybe we can limit these options to just the `dawn_native_*` targets, although I haven't looked at them TBH.

**Pull request merged**